### PR TITLE
DS-6042: update Drupal to 8.6.0 and remove patch for 736066

### PR DESCRIPTION
**Pull request by jaapjan**
## Problem
There is a new Drupal version!

## Solution
Let's update to the newest!

For this I needed to make changes in:
- Patches used in the project.
- Injecting the correct services in PostForm.php.
- Update to dynamic entity reference 1.6.
- Use PHPUnit 6.x (also made change in drupal_social composer.json and enterprise projects in Project Moss) and changing some code in the unit test.
- Behat test changes: mostly waiting for AJAX.
- Getting feature status clean.
- Caching disabled for the anonymous event enrollment.
- Changing the title of the url path settings fieldset
- Use open Social 2.0.0 in install_update job.
- Using latest version of data_policy https://www.drupal.org/project/data_policy/issues/2999425 to prevent conflict with dynamic page cache.

## Issue tracker
- DS-6042
- https://www.drupal.org/project/social/issues/2998671

## HTT
- [x] Check out the code changes
- [x] Check that all the patches apply
- [x] Make sure changes are in the .make files and the composer files.
- [x] Check all the tests.
- [x] Check change records and release notes of D8.6.0 and D8.6.1.
- [ ] Ask me about any questions you might have.

## Release notes
-

### CODE REVIEW

**Comment by jaapjan** [comment]
Failing projects in Moss seems to be false negatives. The notification-mails.feature is solved in pullrequest #1021.

The rest of the failing tests should be working, not sure what is happening there. I've retriggered the tests for SHNL.

File: composer.json [hunk 1]
```diff
@@ -32,10 +32,9 @@
         "patches": {
             "drupal/core": {
                 "Color module html preview optional": "https://www.drupal.org/files/issues/color-optional-html-preview-2844190-2.patch",
-                "Restrict images to this site blocks image style derivatives": "https://www.drupal.org/files/issues/image_restrict_image_styles-2528214-24.patch",
+                "Restrict images to this site blocks image style derivatives": "https://www.drupal.org/files/issues/2018-06-14/2528214-36.patch",
```
Comment (bramtenhove) [inline composer.json]: There is a newer patch available here: https://www.drupal.org/files/issues/2018-06-22/2528214-40.patch.

Comment (jaapjan) [reply composer.json]: Correct, but 40 is not compatible with Drupal 8.6. And 40 is not solving the issue completely as well. See my comment on this issue: https://github.com/goalgorilla/open_social/pull/976 I think it deserves more attention from ITcrowd as well. Especially if we want to get this in for certain enterprise projects.

Comment (bramtenhove) [reply composer.json]: I thought that was indeed the case, but couldn't find anymore where it was said. Check.

**Pull request merged**